pkg/util/push/cortexotlpconverter/metrics_to_prw.go

```diff
@@ -0,0 +1,373 @@
+package cortexotlpconverter
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"sort"
+	"time"
+
+	"github.com/prometheus/otlptranslator"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/util/annotations"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/multierr"
+
+	"github.com/cortexproject/cortex/pkg/cortexpb"
+)
+
+type PromoteResourceAttributes struct {
+	promoteAll bool
+	attrs      map[string]struct{}
+}
+
+type Settings struct {
+	Namespace                         string
+	ExternalLabels                    map[string]string
+	DisableTargetInfo                 bool
+	AddMetricSuffixes                 bool
+	AllowUTF8                         bool
+	PromoteResourceAttributes         *PromoteResourceAttributes
+	KeepIdentifyingResourceAttributes bool
+	ConvertHistogramsToNHCB           bool
+	AllowDeltaTemporality             bool
+	// LookbackDelta is the PromQL engine lookback delta.
+	LookbackDelta time.Duration
+	// PromoteScopeMetadata controls whether to promote OTel scope metadata to metric labels.
+	PromoteScopeMetadata    bool
+	EnableTypeAndUnitLabels bool
```

Comment:
We should clean up the settings a little bit. If there are options that are not in Cortex for now, let's not include them in this PR